File: T3/T3_Dialogos/app/src/main/java/com/system/develop/t3_dialogos/dialogos/DialogoListaSimple.java
package com.system.develop.t3_dialogos.dialogos;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.util.Log;
import android.widget.Toast;

public class DialogoListaSimple extends DialogFragment {

    OnDialogoListaListener listener;
    Context context;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.context = context;
        try{
            listener = (OnDialogoListaListener) context;
        }catch (ClassCastException e){
            Log.v("error","error en el casteo");
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        final String[] opciones = {"sdfsdf","sdfsdfsdf","qrwtghgnfbv"};

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("TITULO LISTA");
        //builder.setMessage("asdasd");
        builder.setItems(opciones, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                //Toast.makeText(getContext(),opciones[which],Toast.LENGTH_SHORT).show();
                listener.onElementoListaSelected(opciones[which]);
            }
        });
        //builder.setMu
        return builder.create();
    }

    public interface OnDialogoListaListener{
        public void onElementoListaSelected(String s);
    }
}

File: Kotlin/T5/codigos2223/T5_Dialogos/app/src/main/java/com/develop/t5_dialogos/dialogs/DialogoListaSimple.kt
package com.develop.t5_dialogos.dialogs

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment

class DialogoListaSimple: DialogFragment() {

    private lateinit var listener: OnListaSimpleListener;
    private  var elementoSeleccionado: String? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        listener = context as OnListaSimpleListener
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder: AlertDialog.Builder = AlertDialog.Builder(requireContext())
        val elementos = arrayOf("Opcion 1", "Opcion 2", "Opcion 3")
        builder.setTitle("Titulo de lista")
        builder.setSingleChoiceItems(elementos, -1,
            DialogInterface.OnClickListener { _, i ->
                elementoSeleccionado = elementos[i]
            })
        builder.setPositiveButton("Aceptar", DialogInterface.OnClickListener { dialogInterface, i ->
            // como comunico la seleccion a la activity
            listener.onListaSelected(elementoSeleccionado)
        })

        builder.setNegativeButton("Cancelar", DialogInterface.OnClickListener { dialogInterface, i ->
            // como comunico la seleccion a la activity
            dismiss()
        })

        return builder.create()
    }


    interface OnListaSimpleListener{
        fun onListaSelected(elemento: String?)
    }
}